viewer/esProxy.js

```diff
@@ -0,0 +1,280 @@
+  }
+
+  if (sensors[credentials.name].pass !== undefined && sensors[credentials.name].pass !== credentials.pass) {
+    console.log(`Incorrect password for ${credentials.name}`);
+    return res.set('WWW-Authenticate', 'Basic').status(401).send();
+  }
+
+  req.sensor = sensors[credentials.name];
+  if (!sensors[credentials.name].ip) {
+    return next();
+  }
+
+  let ip = req.connection.remoteAddress;
+  if (ip.startsWith('::ffff:')) {
+    ip = ip.substring(7);
+  }
+
+  if (!req.sensor.ip.includes(ip)) {
+    console.log(`Incorrect source ip ${ip} for ${credentials.name}`);
+    return res.set('WWW-Authenticate', 'Basic').status(401).send();
+  }
+
+  return next();
+});
+
+// ============================================================================
+// Proxy code to real ES
+// ===========================================================================
+
+// Save the post body
+
+function hasBody (req) {
+  var encoding = 'transfer-encoding' in req.headers;
+  var length = 'content-length' in req.headers && req.headers['content-length'] !== '0';
+  return encoding || length;
+}
+
+function saveBody (req, res, next) {
+  if (req._body) { return next(); }
+  req.body = req.body || {};
+
+  if (!hasBody(req)) { return next(); }
+
+  // flag as parsed
+  req._body = true;
+
+  // parse
+  var buf = Buffer.alloc(parseInt(req.headers['content-length'] || '1024'));
+  let pos = 0;
+  req.on('data', (chunk) => { chunk.copy(buf, pos); pos += chunk.length; });
+  req.on('end', () => {
+    req.body = buf;
+    next();
+  });
+}
+
+// Proxy
+
+function doProxy (req, res, cb) {
+  let result = '';
+  let url = elasticsearch + req.url;
+  console.log('URL', url);
+  var info = URL.parse(url);
+  info.method = req.method;
+  var client;
+  if (url.match(/^https:/)) {
+    info.agent = httpsAgent;
+    client = https;
+  } else {
+    info.agent = httpAgent;
+    client = http;
+  }
+
+  var preq = client.request(info, (pres) => {
+    pres.on('data', (chunk) => {
+      result += chunk.toString();
+    });
+    pres.on('end', () => {
+      res.setHeader('content-type', 'application/json');
+      res.send(result);
+    });
+  });
+  preq.setHeader('content-type', req.headers['content-type'] || 'application/json');
+
+  // Copy these headers from original request to new request
+  for (let header of ['x-opaque-id', 'content-encoding', 'content-length']) {
+    if (req.headers[header] !== undefined) {
+      preq.setHeader(header, req.headers[header]);
+    }
+  }
+
+  preq.on('error', (e) => {
+    console.log(`Request error ${url}`, e);
+  });
+  if (req._body) {
+    preq.end(req.body);
+  } else {
+    preq.end('');
+  }
+}
+
+// ============================================================================
+// handle the requests from the capture/viewer
+// ===========================================================================
+
+// Get requests
+app.get('*', (req, res) => {
+  const path = req.params['0'];
+
+  if (getExact[path]) {
```

Comment:
What is this part all about? Why doesn't each case have anything inside of it? Is it a stub for the future?

Reply:
And why not SWITCH!? 😄

Reply:
Maybe create a function for the parts that are the same for the POST and GET parts?

Reply:
they are empty because they aren't returning an error and will do success at bottom. I'll add a comment